cmd/thanos/store.go

```diff
@@ -67,6 +67,9 @@ func registerStore(app *extkingpin.App) {
 	maxSampleCount := cmd.Flag("store.grpc.series-sample-limit",
 		"Maximum amount of samples returned via a single Series call. The Series call fails if this limit is exceeded. 0 means no limit. NOTE: For efficiency the limit is internally implemented as 'chunks limit' considering each chunk contains 120 samples (it's the max number of samples each chunk can contain), so the actual number of samples might be lower, even though the maximum could be hit.").
 		Default("0").Uint()
+	maxTouchSeriesCount := cmd.Flag("store.grpc.touch-series-limit",
```

Comment:
Amazing. What I miss here is the definition of "touched" (: 

Is this applied before fetching Series ID and chunk metas?
Is this applied while fetchink chunk metas and only if there is chunk matching time range?

This will allow understanding this better for users (:  I see from code it's the former and that's ok for now, however, hopefully, we can improve and touch less series here with this work: https://github.com/thanos-io/thanos/issues/3512

Reply:
yes, Is this before fetching Series ID and chunk.

Reply:
I changed maxTouchSeriesCount to maxTouchedSeriesCount